### PR TITLE
ING-652: Add support for bucket history settings

### PR DESCRIPTION
**Pull request by Westwooo**

### CODE REVIEW

File: cbmgmtx/mgmt.go [hunk 1]
```diff
@@ -623,6 +639,14 @@ func (h Management) decodeMutableBucketSettings(data *bucketSettingsJson) (*Muta
 	settings.MaxTTL = time.Duration(data.MaxTTL) * time.Second
 	settings.CompressionMode = CompressionMode(data.CompressionMode)
 	settings.DurabilityMinLevel = DurabilityLevel(data.MinimumDurabilityLevel)
+	settings.HistoryRetentionBytes = data.HistoryRetentionBytes
+	settings.HistoryRetentionSeconds = data.HistoryRetentionSeconds
+
+	if data.StorageBackend != "magma" {
+		settings.HistoryRetentionCollectionDefault = nil
```
Comment (chvck) [inline cbmgmtx/mgmt.go:646]: This seems like it should be rejected or passed through to be handled server side.

Comment (Westwooo) [reply cbmgmtx/mgmt.go:646]: ns_server always returns a bool value for this field, no matter what the storage type is. So I wanted to differentiate between false and the zero value.

Comment (chvck) [reply cbmgmtx/mgmt.go:646]: I think we should just return the bool that ns_server does, *but* possibly making it a bool pointer in the JSON as older server versions certainly won't return this field at all.

Comment (Westwooo) [reply cbmgmtx/mgmt.go]: Done

**Review by chvck — APPROVED** [review]

**Pull request merged**